### ISSUE TITLE
Investigate & document app version history

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#8003]
To my current understanding:

* Apple forces us to update iOS/Xcode versions, and we choose to update react native (RN) versions
* Both of those actions result in older OS versions not being supported
* When a user is on a device that is no longer supported, their max our-app version becomes capped (on that device) at the latest release that supported their version

This ticket:
* Map out when/where those version strandings have already happened
* Ideally, also get #s on how many people we're talking about
* Document that historical information somewhere (and likely we will update it as we go/continue to do app strandings)

**Comment by TKDickson** [comment]
Relevant notes from [a slack conversation](https://adhoc.slack.com/archives/C02N65QRRB7/p1707494621326149?thread_ts=1707489348.781479&cid=C02N65QRRB7):

> Ok. Just went through on testing it and we will still be able to support our current set of devices for iOS. We just have to update our deployment target to 17 which will force every developer/QA here to upgrade to xcode 15 if they haven’t already, which might have us upgrade to the latest macOS
> 
> When we update RN 0.73 that will the be the last version of RN that support Android 5.0. React Native 0.74 will have a minimum sdk of Android 6.0 and will discontinue whatever support there is for Android devices that can’t go to Android 6.0.
> 
> Currently RN will still support our list of iOS devices, so nothing will change there.
> 
> TLDR is that this update requirement from Apple won’t change what devices are supported. It will soon though, Xcode 15 minimum version is iOS 12. We have a minimum of iOS 13. Previous version supported iOS 11 and the prior version to that supported iOS 9.
> 
> What happens to devices that can’t match our minimum version is that they will still be able to download and use our app, but it is capped at the last version available that supports their device and they just won’t receive further updates.
> 
> Thanks Dylan, that’s good to know.
> If I have questions about it, is it easy to tell when we’ve changed xcode and RN versions over time (like - the entire history of the app)? That might help explain which versions of the app we see people using
> 
> Xcode version not so much. For that one if you have access to the apple developer data to see users iOS version you could loosely tie it to this
> https://developer.apple.com/support/xcode/
> developer.apple.comdeveloper.apple.com
> 
> we currently deploy with version 16.2
> Xcode version 15 is April 29, 2024
> Xcode version 14.1 was April 25, 2023
> Xcode version 13 was April 25, 2022
> I can’t see any history for further back than that

**Comment by TKDickson** [comment]
Alright, stuff in [this google doc](https://docs.google.com/spreadsheets/d/1JLL5xmx_hBpMJckz6RfmaKq_8NQy6XZg4OC7cijGJEk/edit?usp=sharing).

For iOS users in the last 12 months (rolling Feb 14 2023-Feb 14 2024):
For the 86.08% of iOS users where we know the device model they're using***, 1.61% are capped at iOS 15, 2.61% are capped at iOS 16, and the rest (81.86%) are supported iOS 17 and beyond (leaving the 13.92% of unknown).

However, in the last 12 months rolling, we've had users on older iOS versions:
0.08% iOS 13 (1546)
0.51% iOS 14 (9840)
Remaining 99.4% iOS15+

All of the iOS 13 are by choice (they're using known devices, whose max supported iOS versions are higher than 13). The majority of the iOS 14 are also choice -- although 422 (~4.3%) are in the "we don't know their max device version" category. 


***13.92% of users I can't tell you what device model they're using, with just Firebase. There's a decent chance that a query in big query could [get us the details for a good chunk of them](https://stackoverflow.com/questions/58937511/in-firebase-analytics-one-of-the-devices-listed-is-just-iphone-what-does-thi). But I can't run that with the current issues with financials and big query.

**Comment by TKDickson** [comment]
Android is..... more difficult to get a map of the landscape. Mainly because I can only pull from Firebase so my ability to quickly work magic (against just under 3,100 different Android-running device models in use in the past 12 months) is limited.

What I can say is that it doesn't _seem_ like anyone is prevented from using our app with Android. The rate-limiting step with Android is our react native (RN) version.
> When we update RN 0.73 that will the be the last version of RN that support Android 5.0. React Native 0.74 will have a minimum sdk of Android 6.0 and will discontinue whatever support there is for Android devices that can’t go to Android 6.0.

There are not any users (in the last 12 months rolling) on an Android version lower than Android 8 (0.39% of total Android users), and the bulk of Android users (~87%) are on Android 12 or higher. So no one is prevented from updating our-app version by their Android OS, yet.

**Comment by TKDickson** [comment]
One thing I'll also jot down because I want to include it in final documentation (and likely I can find writing on the internet more comprehensive and snappy than my ideas) - there are plenty of reasons (don't want to, don't know how, know how & want to but don't have enough space, etc) to not update OS even if you could technically update OS by manufacturer limits.

**Comment by TKDickson** [comment]
Argument to be made that Android 10 is more problematic for our app/users, from poking around in Google Play Console:
<img width="1333" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWhwQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3467f2dcd46b72e4f9b8cd2f0299f870a1372584/image.png" alt="image.png" />

**Comment by TKDickson** [comment]
Our minSDK version also hasn't changed in a long time (since the thing I referenced in my original sprint 0), so no changes to version support there:
```
buildscript {
    ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 26
        compileSdkVersion = 33
        targetSdkVersion = 33
        KOTLIN_VERSION = "1.8.0"
        kotlinVersion = "1.8.0"
```

**Comment by TKDickson** [comment]
I assume we would continue to [document app support history on this page](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/product/significant%20decisions.md#decision-6-mobile-os-support-which-mobile-operating-systems-will-the-app-run-on).

The notes in this ticket are way more detailed than that format would support; leaving them ticket-only until/unless we pick a different documentation strategy and location for these decisions.

The short version: no actual changes in what we've needed to support since last decision in June 2021 (phew!). However the horizon for this is going to creep up for iOS, so should be prepared with decision-making (and how enforcing) proactively, not reactively.
